### ISSUE TITLE
Path must be canonical but got: '//..'

### ISSUE THREAD

**Issue by blacksmoke26** [klesun/deep-js-completion#13]
```
Worker exited due to exception

java.lang.IllegalArgumentException: path must be canonical but got: '//..'
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl$FsRoot.<init>(PersistentFSImpl.java:1413)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl$FsRoot.<init>(PersistentFSImpl.java:1400)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl.findRoot(PersistentFSImpl.java:1028)
	at com.intellij.openapi.vfs.newvfs.persistent.PersistentFSImpl.findRoot(PersistentFSImpl.java:57)
	at com.intellij.openapi.vfs.newvfs.VfsImplUtil.prepare(VfsImplUtil.java:139)
	at com.intellij.openapi.vfs.newvfs.VfsImplUtil.findFileByPath(VfsImplUtil.java:38)
	at com.intellij.openapi.vfs.impl.local.LocalFileSystemBase.findFileByPath(LocalFileSystemBase.java:46)
	at org.klesun.deep_js_completion.entry.PathStrGoToDecl$.$anonfun$getReferencedFileStrict$4(PathStrGoToDecl.scala:23)
	at scala.Option.flatMap(Option.scala:188)
	at org.klesun.deep_js_completion.entry.PathStrGoToDecl$.getReferencedFileStrict(PathStrGoToDecl.scala:23)
	at org.klesun.deep_js_completion.entry.PathStrGoToDecl$.getReferencedFileLoose(PathStrGoToDecl.scala:29)
	at org.klesun.deep_js_completion.entry.PathStrGoToDecl$.$anonfun$getReferencedFileLoose$4(PathStrGoToDecl.scala:48)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:488)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:489)
	at scala.collection.Iterator$ConcatIterator.hasNext(Iterator.scala:222)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:488)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:488)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:488)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$ConcatIterator.advance(Iterator.scala:198)
	at scala.collection.Iterator$ConcatIterator.hasNext(Iterator.scala:225)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:488)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:511)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2(DeepJsLang.scala:258)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2$adapted(DeepJsLang.scala:258)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:513)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$It.lift(DeepJsLang.scala:232)
	at org.klesun.lang.DeepJsLang$.frs(DeepJsLang.scala:259)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:81)
	at org.klesun.deep_js_completion.contexts.ExprCtx.findExprType(ExprCtx.scala:35)
	at org.klesun.deep_js_completion.resolvers.FuncCallRes.$anonfun$resolve$1(FuncCallRes.scala:67)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:511)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2(DeepJsLang.scala:258)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2$adapted(DeepJsLang.scala:258)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:513)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$It.lift(DeepJsLang.scala:232)
	at org.klesun.lang.DeepJsLang$.frs(DeepJsLang.scala:259)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:81)
	at org.klesun.deep_js_completion.contexts.ExprCtx.findExprType(ExprCtx.scala:35)
	at org.klesun.deep_js_completion.resolvers.VarRes.$anonfun$resolveDestructEl$1(VarRes.scala:149)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at scala.collection.Iterator$ConcatIterator.hasNext(Iterator.scala:222)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:488)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:489)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:488)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$ConcatIterator.advance(Iterator.scala:198)
	at scala.collection.Iterator$ConcatIterator.hasNext(Iterator.scala:225)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:488)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$ConcatIterator.advance(Iterator.scala:198)
	at scala.collection.Iterator$ConcatIterator.hasNext(Iterator.scala:225)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:511)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2(DeepJsLang.scala:258)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2$adapted(DeepJsLang.scala:258)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:513)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$It.lift(DeepJsLang.scala:232)
	at org.klesun.lang.DeepJsLang$.frs(DeepJsLang.scala:259)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:81)
	at org.klesun.deep_js_completion.contexts.ExprCtx.findExprType(ExprCtx.scala:35)
	at org.klesun.deep_js_completion.resolvers.FuncCallRes.$anonfun$resolve$1(FuncCallRes.scala:67)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:511)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2(DeepJsLang.scala:258)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2$adapted(DeepJsLang.scala:258)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:513)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$It.lift(DeepJsLang.scala:232)
	at org.klesun.lang.DeepJsLang$.frs(DeepJsLang.scala:259)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:81)
	at org.klesun.deep_js_completion.contexts.ExprCtx.findExprType(ExprCtx.scala:35)
	at org.klesun.deep_js_completion.resolvers.MainRes$.$anonfun$resolveIn$18(MainRes.scala:144)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:489)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:511)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2(DeepJsLang.scala:258)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2$adapted(DeepJsLang.scala:258)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:513)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$It.lift(DeepJsLang.scala:232)
	at org.klesun.lang.DeepJsLang$.frs(DeepJsLang.scala:259)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:81)
	at org.klesun.deep_js_completion.contexts.ExprCtx.findExprType(ExprCtx.scala:35)
	at org.klesun.deep_js_completion.resolvers.VarRes.$anonfun$resolveMainDeclVar$1(VarRes.scala:159)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at scala.collection.Iterator$ConcatIterator.hasNext(Iterator.scala:222)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:488)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$ConcatIterator.advance(Iterator.scala:198)
	at scala.collection.Iterator$ConcatIterator.hasNext(Iterator.scala:225)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:511)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2(DeepJsLang.scala:258)
	at org.klesun.lang.DeepJsLang$.$anonfun$frs$2$adapted(DeepJsLang.scala:258)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:513)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$It.lift(DeepJsLang.scala:232)
	at org.klesun.lang.DeepJsLang$.frs(DeepJsLang.scala:259)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:81)
	at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:49)
	at org.klesun.deep_js_completion.completion_providers.PropNamePvdr.$anonfun$getGotoDeclarationTargets$4(PropNamePvdr.scala:237)
	at scala.collection.Iterator$$anon$11.nextCur(Iterator.scala:484)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:490)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:489)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:489)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:511)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:488)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:489)
	at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:205)
	at org.klesun.lang.DeepJsLang$It.lift(DeepJsLang.scala:232)
	at org.klesun.deep_js_completion.completion_providers.PropNamePvdr.getGotoDeclarationTargets(PropNamePvdr.scala:243)
	at com.intellij.codeInsight.navigation.actions.GotoDeclarationAction.findTargetElementsFromProviders(GotoDeclarationAction.java:274)
	at com.intellij.codeInsight.navigation.actions.GotoDeclarationAction.findTargetElementsNoVS(GotoDeclarationAction.java:285)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler.getInfoAt(CtrlMouseHandler.java:478)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler.getInfoAt(CtrlMouseHandler.java:457)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler.access$1500(CtrlMouseHandler.java:93)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$TooltipProvider.doExecute(CtrlMouseHandler.java:738)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$TooltipProvider.access$2100(CtrlMouseHandler.java:655)
	at com.intellij.codeInsight.navigation.CtrlMouseHandler$TooltipProvider$1.performInReadAction(CtrlMouseHandler.java:709)
	at com.intellij.openapi.progress.util.ReadTask.lambda$runBackgroundProcess$0(ReadTask.java:66)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:974)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:57)
	at com.intellij.openapi.progress.util.ReadTask.runBackgroundProcess(ReadTask.java:66)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runUnderProgress$4(ProgressIndicatorUtils.java:225)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$3(CoreProgressManager.java:180)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:586)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:86)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:180)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runUnderProgress(ProgressIndicatorUtils.java:223)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.access$100(ProgressIndicatorUtils.java:33)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils$4.run(ProgressIndicatorUtils.java:174)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Info
```
PhpStorm 2019.1
Build #PS-191.6183.55, built on March 20, 2019
JRE: 1.8.0_202-release-1483-b39 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 8.1 6.3
---
Plugin Version: 2019-02-19-0001
```

**Comment by klesun** [comment]
Wow, someone actually uses the js pluign =-D
Thanks, I'll take a look as soon as I can.

**Comment by klesun** [comment]
The fix - [2019.03.28.001](https://plugins.jetbrains.com/plugin/11478-deep-js-completion/update/60121)
Few working days will pass before it's published as JetBrains now verifies all plugin updates.

**Comment by brxkensmile** [comment]
> Wow, someone actually uses the js pluign =-D
> Thanks, I'll take a look as soon as I can.

@klesun I'm using your plugin too!1 Now I'm offended!!
